packages/eas-cli/src/project/discourageExpoGoForProd.ts

```diff
@@ -0,0 +1,73 @@
+import { Workflow } from '@expo/eas-build-job';
+import chalk from 'chalk';
+import getenv from 'getenv';
+import resolveFrom from 'resolve-from';
+
+import { resolveWorkflowPerPlatformAsync } from './workflow';
+import Log, { learnMore } from '../log';
+import type { ProfileData } from '../utils/profiles';
+import type { Client } from '../vcs/vcs';
+
+const suppressionEnvVarName = 'EAS_BUILD_NO_EXPO_GO_WARNING';
+
+export const discourageExpoGoForProd = (
```

Comment:
I would not try to do anything smart here with the non-awaited Promise. Can we just await this? Does this add significant time to the build process?

If we do this asynchronously it might happen that a chain of ~thought~ log may be broken by this warning.